internal/bpf/bpfWrapper.c

```diff
@@ -190,14 +191,14 @@ int Load_attach_bpf_xdp_pass(char *ifname) {
 		 __FUNCTION__, ifname, ifindex);
 
 	/* Load the BPF program */
-	err = bpf_prog_load(filename, BPF_PROG_TYPE_XDP, &obj, &prog_fd);
+	err = bpf_xdp_query_id(ifindex, (int)xdp_flags, &prog_fd);

```

Comment:
I don't think this is loading the BPF program. This function looks like it uses the ifindex to query and see if there's any program already loaded on an in interface. Note that you aren't actually doing anything with the file `/afxdp/xdp_pass.o`. I think i sent a pointer in a previous comment on how to load with libxdp from a file. Happy to write this up if needed. 

but bpf_xdp_query_id is only querying the ifindex to see if there's something already loaded.